django/utils/autoreload.py

```diff
@@ -242,96 +294,27 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-def raise_last_exception():
-    global _exception
-    if _exception is not None:
-        raise _exception[1]
+def get_reloader(use_watchman=False):
+    if use_watchman and WatchmanReloader.is_available():
+        return WatchmanReloader()
 
+    return StatReloader()
 
-def ensure_echo_on():
-    if termios:
-        fd = sys.stdin
-        if fd.isatty():
-            attr_list = termios.tcgetattr(fd)
-            if not attr_list[3] & termios.ECHO:
-                attr_list[3] |= termios.ECHO
-                if hasattr(signal, 'SIGTTOU'):
-                    old_handler = signal.signal(signal.SIGTTOU, signal.SIG_IGN)
-                else:
-                    old_handler = None
-                termios.tcsetattr(fd, termios.TCSANOW, attr_list)
-                if old_handler is not None:
-                    signal.signal(signal.SIGTTOU, old_handler)
 
+def run_with_reloader(main_func, *args, **kwargs):
+    import signal
+    signal.signal(signal.SIGTERM, lambda *args: sys.exit(0))
 
-def reloader_thread():
-    ensure_echo_on()
-    if USE_INOTIFY:
-        fn = inotify_code_changed
-    else:
-        fn = code_changed
-    while RUN_RELOADER:
-        change = fn()
-        if change == FILE_MODIFIED:
-            sys.exit(3)  # force reload
-        elif change == I18N_MODIFIED:
-            reset_translations()
-        time.sleep(1)
-
-
-def restart_with_reloader():
-    import django.__main__
-    while True:
-        args = [sys.executable] + ['-W%s' % o for o in sys.warnoptions]
-        if sys.argv[0] == django.__main__.__file__:
-            # The server was started with `python -m django runserver`.
-            args += ['-m', 'django']
-            args += sys.argv[1:]
-        else:
-            args += sys.argv
-        new_environ = os.environ.copy()
-        new_environ["RUN_MAIN"] = 'true'
-        exit_code = subprocess.call(args, env=new_environ)
-        if exit_code != 3:
-            return exit_code
+    with contextlib.suppress(KeyboardInterrupt):
```

Comment:
Please replace this with a regular `try/except`.

Reply:
Why?

Reply:
This was a pattern that was added and then removed at some point. Not sure about the story behind, but I prefer try/except over `contextlib.suppress`, not sure why I added it here.

Reply:
It's slower, about 10 times.

Reply:
For the interested, it was reverted in PR #9038 because it was [slower](https://code.djangoproject.com/ticket/27818#comment:7), likely due to the overhead of an additional [function call](https://code.djangoproject.com/ticket/27818#comment:10).